lib/usb/Makefile

```diff
@@ -1,32 +1,59 @@
-USB_SRCS := \
-	$(NXDK_DIR)/lib/usb/host/ohci-hcd.c \
-	$(NXDK_DIR)/lib/usb/core/message.c \
-	$(NXDK_DIR)/lib/usb/core/hcd.c \
-	$(NXDK_DIR)/lib/usb/core/hcd-pci.c \
-	$(NXDK_DIR)/lib/usb/core/hub.c \
-	$(NXDK_DIR)/lib/usb/core/usb.c \
-	$(NXDK_DIR)/lib/usb/core/config.c \
-	$(NXDK_DIR)/lib/usb/core/urb.c \
-	$(NXDK_DIR)/lib/usb/core/buffer_simple.c \
-	$(NXDK_DIR)/lib/usb/core/usb-debug.c \
-	$(NXDK_DIR)/lib/usb/sys/BootUSB.c \
-	$(NXDK_DIR)/lib/usb/sys/linuxwrapper.c \
-	$(NXDK_DIR)/lib/usb/sys/xpad.c \
-	$(NXDK_DIR)/lib/usb/sys/xremote.c \
-	$(NXDK_DIR)/lib/usb/sys/usbkey.c \
-	$(NXDK_DIR)/lib/usb/sys/usbmouse.c \
-	$(NXDK_DIR)/lib/usb/misc/misc.c \
-	$(NXDK_DIR)/lib/usb/misc/pci.c \
-	$(NXDK_DIR)/lib/usb/misc/malloc.c
-
-USB_OBJS = $(addsuffix .obj, $(basename $(USB_SRCS)))
-
-$(NXDK_DIR)/lib/libnxdk_usb.lib: $(USB_OBJS)
-
-main.exe: $(NXDK_DIR)/lib/libnxdk_usb.lib
-
-CLEANRULES += clean-usb
-
-.PHONY: clean-usb
-clean-usb:
-	$(VE)rm -f $(USB_OBJS) $(NXDK_DIR)/lib/libnxdk_usb.lib
+USBH_DIR=$(NXDK_DIR)/lib/usb/ohci_usbhostlib
+
+#USB Host core
+USB_CORE_SRCS := \
+	$(USBH_DIR)/src_core/hub.c \
+	$(USBH_DIR)/src_core/mem_alloc.c \
+	$(USBH_DIR)/src_core/ohci.c \
+	$(USBH_DIR)/src_core/support.c \
+	$(USBH_DIR)/src_core/usb_core.c
+
+#USB Human Input Device class driver
+USB_HID_SRCS := \
+	$(USBH_DIR)/src_hid/hid_core.c \
+	$(USBH_DIR)/src_hid/hid_driver.c
+
+#USB Communication Device class driver
+USB_CDC_SRCS := \
+	$(USBH_DIR)/src_cdc/cdc_core.c \
+	$(USBH_DIR)/src_cdc/cdc_driver.c \
+	$(USBH_DIR)/src_cdc/cdc_parser.c
+	
+#USB Audio class driver
+USB_UAC_SRCS := \
+	$(USBH_DIR)/src_uac/uac_core.c \
+	$(USBH_DIR)/src_uac/uac_driver.c \
+	$(USBH_DIR)/src_uac/uac_parser.c
+
+# USB Video class driver
+USB_UVC_SRCS := \
+	$(USBH_DIR)/src_uvc/uvc_core.c \
+	$(USBH_DIR)/src_uvc/uvc_driver.c \
+	$(USBH_DIR)/src_uvc/uvc_parser.c

```

Comment:
- Did you test these?
- Can we make these optional? (For smaller binaries + faster compilation + promoting the idea of having modular drivers)

Reply:
Audio is tested, video has not. I dont have a webcam or similar at all. It has some similaries to audio devices and uses isochronous endpoints too.

Reply:
I modified the makefile so that all class drivers are optional. HID are enabled by default, all others are enabled in a similar way to how SDL2 is enabled in nxdk.

`NXDK_USB_ENABLE_CDC=y` in makefile will enable CDC class driver for example.

I would image the user expects HID to work out of the box so this is enabled by default but can be disabled by `NXDK_USB_DISABLE_HID=y`

I wasnt really sure if this is a good implementation. If you had anything in mind let me know.